src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -87,130 +97,139 @@ public function handle(AddProductToOrderCommand $command)
     {
         $order = $this->getOrderObject($command->getOrderId());
 
-        if ($this->context->cart !== null) {
-            $oldCartRules = $this->context->cart->getCartRules();
-        } else {
-            $oldCartRules = [];
-        }
+        $this->contextStateManager
+            ->setCurrency(new Currency($order->id_currency))
+            ->setCustomer(new Customer($order->id_customer))
+        ;
 
-        $this->assertOrderWasNotShipped($order);
+        try {
+            $oldCartRules = (new Cart($order->id_cart))->getCartRules();
 
-        $product = $this->getProductObject($command->getProductId(), (int) $order->id_lang);
-        $combination = $this->getCombination($command->getCombinationId());
+            $this->assertOrderWasNotShipped($order);
```

Comment:
May be you can up this before $oldCartRules.

That permit to not assign value to variables if order was not shipped.

Reply:
done